Applications/Spire/Source/BookView/TechnicalsPanel.cpp

```diff
@@ -0,0 +1,208 @@
+  auto fields = std::vector<TextBox*>{
+    make_technicals_value_field<Money>(m_technicals,
+      &SecurityTechnicals::m_high),
+    make_technicals_value_field<Money>(m_technicals,
+      &SecurityTechnicals::m_low),
+    make_technicals_value_field<Money>(m_technicals,
+      &SecurityTechnicals::m_open),
+    make_technicals_value_field<Money>(m_technicals,
+      &SecurityTechnicals::m_close),
+    make_technicals_value_field<Quantity>(m_technicals,
+      &SecurityTechnicals::m_volume),
+    m_default_field};
+  for(auto i = 0; i < std::ssize(fields); ++i) {
+    link(*this, *name_indicators[i]);
+    link(*this, *short_name_indicators[i]);
+    link(*this, *fields[i]);
+  }
+  auto adaptive_box = new AdaptiveBox();
+  adaptive_box->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Fixed);
+  adaptive_box->add(*make_large_layout(name_indicators, fields));
+  adaptive_box->add(*make_large_layout(short_name_indicators, fields));
+  adaptive_box->add(*make_small_layout(name_indicators, fields));
+  m_extra_small_layout = make_small_layout(short_name_indicators, fields);
+  adaptive_box->add(*m_extra_small_layout);
+  auto body = new QWidget();
+  enclose(*body, *adaptive_box);
+  auto box = new Box(body);
+  box->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Fixed);
+  enclose(*this, *box);
+  proxy_style(*this, *box);
+  m_style_connection = connect_style_signal(*this,
+    std::bind_front(&TechnicalsPanel::on_style, this));
+  update_style(*this, [] (auto& style) {
+    style.get(Any()).
+      set(BackgroundColor(QColor(0xFFFFFF))).
+      set(horizontal_padding(scale_width(8))).
+      set(vertical_padding(scale_height(4)));
+    style.get(Any() > is_a<TextBox>()).
+      set(Font(LABEL_FONT()));
+  });
+}
+
+const std::shared_ptr<SecurityTechnicalsModel>&
+    TechnicalsPanel::get_technicals() const {
+  return m_technicals;
+}
+
+const std::shared_ptr<QuantityModel>&
+    TechnicalsPanel::get_default_bid_quantity() const {
+  return m_bid_quantity;
+}
+
+const std::shared_ptr<QuantityModel>&
+    TechnicalsPanel::get_default_ask_quantity() const {
+  return m_ask_quantity;
+}
+
+QSize TechnicalsPanel::minimumSizeHint() const {
+  if(m_minimum_size_hint) {
+    return *m_minimum_size_hint;
+  }
+  m_minimum_size_hint.emplace(m_extra_small_layout->totalMinimumSize().grownBy(
+    get_content_margins(m_geometry)));
+  return *m_minimum_size_hint;
+}
+
+void TechnicalsPanel::on_bid_quantity_update(Quantity quantity) {
+  m_default_field->get_current()->set(
+    to_default_quantity(quantity, m_ask_quantity->get()));
+}
+
+void TechnicalsPanel::on_ask_quantity_update(Quantity quantity) {
+  m_default_field->get_current()->set(
+    to_default_quantity(m_bid_quantity->get(), quantity));
+}
+
+void TechnicalsPanel::on_style() {
+  auto& stylist = find_stylist(*this);
+  for(auto& property : stylist.get_computed_block()) {
+    apply(property, m_geometry, stylist);
+  }
+  m_minimum_size_hint = none;
```

Comment:
I believe when the size hint changes a call to ```updateGeometry``` is also needed. Verify if that's correct.

Reply:
Yes, I think it should be called. Although the official documentation states to call this function only when sizeHint() or sizePolicy() changes, other sources suggest calling it when minimumSizeHint() changes as well. To be safe, it's better to call it in this case.